client/app/private-database/database/add/private-database-database-add.controller.js

```diff
@@ -1,29 +1,31 @@
 angular.module("App").controller(
     "PrivateDatabaseAddBddCtrl",
     class PrivateDatabaseAddBddCtrl {
-        constructor ($scope, $q, $stateParams, Alerter, PrivateDatabase, Hosting) {
+        constructor ($scope, $q, $stateParams, Alerter, PrivateDatabase, Hosting, WhitelistService, Validator) {
             this.$scope = $scope;
             this.$q = $q;
             this.$stateParams = $stateParams;
             this.Alerter = Alerter;
             this.PrivateDatabase = PrivateDatabase;
             this.Hosting = Hosting;
-
-            this.bdd = {
-                value: null,
-                addUser: false,
-                addUserLogin: "",
-                addUserGrant: "",
-                addUserPassword: "",
-                addUserPasswordConfirm: ""
-            };
+            this.whitelistService = WhitelistService;
+            this.validator = Validator;
         }
 
         $onInit () {
             this.productId = this.$stateParams.productId;
 
+            if (this.$scope.isDBaaS()) {
+                this.checkAuthorizedIp()
+                    .then((hasAuthorizedIp) => {
+                        this.model.addIp = !hasAuthorizedIp;
```

Comment:
Maybe you could rename `addIp` to something more explicit like `userCanAddIPAddress`

Reply:
*addIp* is a boolean linked to the checkbox, in the same style as *addUser*. It does not express if a user can or cannot add an ip, but whether he wants or does not want to add one.
this first initialization means that by default, he wants (we suggest him when the forms opens up).
I'll add a comment to clarify that.